### ISSUE TITLE
After the game is updated injection collapse

### ISSUE THREAD

**Issue by potuo** [danielkrupinski/Osiris#3246]
After the game is updated injection collapse

**Comment by potuo** [comment]
Operating system：ArchLinux

**Comment by TauCSGO** [comment]
> Operating system：ArchLinux

Windows too

**Comment by invakid404** [comment]
Crashes for me as well, Ubuntu Focal. Straight segmentation fault when joining a MM match.

**Comment by ReShout876** [comment]
Crashes for me too
Ubuntu Focal 20.04
crashes only when I am about to get spawned in the map.
It is thanks to the Inventory Changer of smt?

**Comment by mdilai** [comment]
Okay, i've debugged this and found issue - incorrect lineGoesThroughSmoke signature.
Quick & dirty workaround to disable smoke check:
```diff --git a/Osiris/Memory.cpp b/Osiris/Memory.cpp
index 57ce009..5a10c61 100644
--- a/Osiris/Memory.cpp
+++ b/Osiris/Memory.cpp
@@ -20,6 +20,7 @@
 #include "Interfaces.h"
 #include "Memory.h"
 #include "SDK/LocalPlayer.h"
+#include "SDK/Vector.h"
 
 template <typename T>
 static constexpr auto relativeToAbsolute(uintptr_t address) noexcept
@@ -237,7 +238,7 @@ Memory::Memory() noexcept
     weaponSystem = *relativeToAbsolute<WeaponSystem**>(findPattern(CLIENT_DLL, "\x48\x8B\x58\x10\x48\x8B\x07\xFF\x10") + 12);
 
     isOtherEnemy = relativeToAbsolute<decltype(isOtherEnemy)>(findPattern(CLIENT_DLL, "\xE8????\x84\xC0\x44\x89\xE2") + 1);
-    lineGoesThroughSmoke = reinterpret_cast<decltype(lineGoesThroughSmoke)>(findPattern(CLIENT_DLL, "\x40\x0F\xB6\xFF\x55"));
+    lineGoesThroughSmoke = [](Vector, Vector, short) noexcept { return false; };
     getDecoratedPlayerName = relativeToAbsolute<decltype(getDecoratedPlayerName)>(findPattern(CLIENT_DLL, "\xE8????\x8B\x33\x4C\x89\xF7") + 1);
 
     hud = relativeToAbsolute<decltype(hud)>(findPattern(CLIENT_DLL, "\x53\x48\x8D\x3D????\x48\x83\xEC\x10\xE8") + 4);
```

But i have a little knowledge how to find new correct signature. @danielkrupinski look at this and also please tell what tools you originally used to get linux signatures?

**Comment by danielkrupinski** [comment]
@mdilai I use Interactive Disassembler but Ghidra should work too.

**Comment by mdilai** [comment]
I've made PR with fix #3265

**Comment by ReShout876** [comment]
I git pulled, a file was modified, recompiled, now it doesnt crash!
